llvm/lib/Target/RISCV/RISCVInsertVSETVLI.cpp

```diff
@@ -478,6 +480,8 @@ class VSETVLIInfo {
     State = AVLIsImm;
   }
 
+  void setAVLDefMI(const MachineInstr *DefMI) { AVLDefMI = DefMI; }
```

Comment:
Can we go further and remove the `AVLIsReg` state entirely? I think we have the invariant that the AVL should be one of three types: 

- A virtual register where we can get the definition
- X0, which is really just VLMAX
- An immediate

So I was thinking we could make the state something like

```c++
enum : uint8_t {
    Uninitialized,
    AVLIsDef,
    AVLIsVLMAX
    AVLIsImm,
    Unknown,
  } State = Uninitialized;
```

Reply:
I've done this part in #89964, but with the existing virtual register form